api4/status.go

```diff
@@ -127,7 +127,7 @@ func updateUserCustomStatus(c *Context, w http.ResponseWriter, r *http.Request)
 	}
 
 	customStatus := model.CustomStatusFromJson(r.Body)
-	if customStatus == nil || (customStatus.Text == "" && customStatus.Emoji == "") {
+	if customStatus == nil || customStatus.Emoji == "" || !customStatus.IsDurationValid() || !customStatus.IsExpirationTimeValid() {
```

Comment:
1. What about the custom-status text?
2. Refactor this to `isValidCustomStatus` variable.

Reply:
Custom status text can be empty.